src/twisted/application/internet.py

```diff
@@ -831,8 +841,8 @@ def _unawait(self, value):
     _init.upon(start, enter=_connecting,
                outputs=[_connect])
     _init.upon(stop, enter=_stopped,
-               outputs=[],
-               collector=lambda _: succeed(None))
+               outputs=[_deferredSucceededWithNone],
```

Comment:
I added it to confirm that all `stop`s have outputs.  I found it made the state chart clearer, too:

![image](https://cloud.githubusercontent.com/assets/704247/22052008/2dc63146-dcfc-11e6-9ce0-d32a8c7fd035.png)